lom/src/plugins/plugins_common/plugins_common.go

```diff
@@ -8,63 +8,177 @@ package plugins_common
 import (
     "lom/src/lib/lomcommon"
     "lom/src/lib/lomipc"
+    "sync"
     "time"
 )
 
-// Plugin interface is implemented by all plugins and is used by plugin manager to call plugin methods
+/*
+ *Plugin interface is implemented by all plugins and is used by plugin manager to call plugin methods
+ */
 type Plugin interface {
     Init(actionCfg *lomcommon.ActionCfg_t) error
     Request(hbchan chan PluginHeartBeat, request *lomipc.ActionRequestData) *lomipc.ActionResponseData
     Shutdown() error
     GetPluginID() PluginId
 }
 
-// TODO: Goutham : Clean up unnecessary fields
-// PluginStage indicates the current stage of plugin. Based on  this value plugin manager decisions. For e.g.  whether to accept requests from engine or not
+/*
+ * PluginStage indicates the current stage of plugin. Based on  this value plugin manager decisions. For e.g.  whether to accept requests from engine or not
+ */
 type PluginStage int
 
 const (
     PluginStageUnknown PluginStage = iota // default value
     PluginStageLoadingSuccess
     PluginStageRequestStarted
-    PluginStageRequestStartedHB // for long running
-    PluginStageRequestError
     PluginStageRequestSuccess
+    PluginStageRequestTimeout
+    PluginStageShutdownStarted
+    PluginStageShutdownCompleted
+    PluginStageShutdownTimeout
+    PluginStageDisabled
+)
+
+/*
```

Comment:
Just a loud thought
Declare last entry as PluiginStageCount
You could declare a slice of strings of size PluginStageCount with hard coded strings.
yourslice = [pluginStageCount]string { ....}
GetPluginStageToString(stage) { return yourSlice[stage] }

Reply:
Yes. Did

Reply:
made the code changes to mark plugin request success after plugin manager sends data to engine. SO lock is not needed. Updated  code.